src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  const adder = (number) => {
+    if (number !== undefined) {
+      result += number;
+
+      return adder;
+    } else {
+      const currentResult = result;
+
+      result = 0;
+
+      return currentResult;
+    }
```

Comment:
```suggestion
    } 
    const currentResult = result;

    result = 0;

    return currentResult;
```
You don't need else as you return in the `if` block